modules/Bio/EnsEMBL/Compara/Utils/Test.pm

```diff
@@ -0,0 +1,113 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::Utils::Test
+
+=head1 DESCRIPTION
+
+Utility functions used in test scripts
+
+=cut
+
+package Bio::EnsEMBL::Compara::Utils::Test;
+
+use strict;
+use warnings;
+
+use Cwd qw(abs_path);
+use File::Spec;
+use File::Basename qw/dirname/;
+
+=head2 GLOBAL VARIABLES
+
+=over
+
+=item repository_root
+
+The path to the root of the repository. Kept here so that we don't need
+to "compute" it again and again.
+
+=back
+
+=cut
+
+my $repository_root;
+
+=head2 get_repository_root
+
+  Description : Return the path to the root of the repository. Note that
+                this is constructed from the path to this module.
+
+=cut
+
+sub get_repository_root {
+    return $repository_root if $repository_root;
+    my $file_dir = dirname(abs_path(__FILE__));
+    if ($file_dir =~ /^(.*\/ensembl-compara)\b/) {
+        $repository_root = $1;
+        return $repository_root;
+    } else {
+        die "Can't find a parent directory named 'ensembl-compara' in '$file_dir'";
+    }
```

Comment:
I use several checkouts of compara in development. They can be named things like `ensembl-compara-master` or `ensembl-compara-release`. While I'm able to just change this regex in each one, I would prefer not to have to..

I have a script that performs exactly this function which continually trims the last dir from a path until `-e $dir/.git` - could we use this means of finding the repo root? (I see you've already rewritten this method once, so no problem if you don't want to do it again)

Reply:
I must confess I haven't been able to find any solution that I'm 100% happy with. If I was to run your suggestion in my computer, my unison is not copying the `.git` folders, so it wouldn't work...
To make it work for you too, we could just update the regex:
```
    if ($file_dir =~ /^(.*\/ensembl-compara[^\\\/])\b/) {
```

Reply:
I'm not 100% happy with this implementation either. Maybe I could be looking for `modules` and consider the parent directory to be the root ? That would work for you @carlacummins ?

Reply:
Actually, that's a good point @muffato . Most things wouldn't work if that is manually modified by users... I like it :)

Reply:
Yes, that would work well! 👍

Reply:
What about reverting to the initial implementation (go 5 directories up) but also add a unit-test checking that the directory returned by the function does contain say `modules`, `scripts` and `sql`. So if we move the function and break the test we would know ? That would also answer this comment:
> Would it be useful to have a unit test for the new `Utils::Test` module? (I see coveralls is not happy)

Reply:
Killing two birds with one stone - nice 🥇